velox/connectors/hive/FileHandle.h

```diff
@@ -67,14 +67,35 @@ using FileHandleCache = SimpleLRUCache<std::string, FileHandle>;
 
 // Creates FileHandles via the Generator interface the CachedFactory requires.
 class FileHandleGenerator {
+ public:
+  struct Stats {
+    uint64_t minGenerationTimeMcrs;
+    uint64_t maxGenerationTimeMcrs;
+    uint64_t totalGenerationTimeMcrs;
+    size_t numGeneration;
+  };
+
  public:
   FileHandleGenerator() {}
   FileHandleGenerator(std::shared_ptr<const Config> properties)
       : properties_(std::move(properties)) {}
   std::unique_ptr<FileHandle> operator()(const std::string& filename);
+  Stats stats() {
+    std::lock_guard<std::mutex> l(m_);
+    return {
+        minGenerationTimeMcrs_,
+        maxGenerationTimeMcrs_,
+        totalGenerationTimeMcrs_,
+        numGeneration_};
+  }
 
  private:
   const std::shared_ptr<const Config> properties_;
+  std::mutex m_;
```

Comment:
If we use m_ only for stats, should we call it statsMutex_?

Reply:
will be gone, too